### ISSUE TITLE
More enhancements for CompositeObservable

### ISSUE THREAD

**Issue by thomasnield** [ReactiveX/RxJavaFX#39]
I won't make these changes right now. Just documenting them. 

~~\* Add vararg `add()` and `remove()` methods~~
~~\* Add `clear()` method to remove all Observables~~
~~\* Back with `ObservableSet` instead of `ObservableList` to prevent duplicates~~
~~\* Add `getBackingList()` to return unmodifiable `ObservableList<Observable<T>>` for currently tracked Observables.~~

**Comment by thomasnield** [comment]
The use of `initialCapacity` might be broken too as the caching happens to each independent subscription. Client might as well cache it themselves with that behavior. But the desired behavior is to have caching happen for all subscribers.

**Comment by thomasnield** [comment]
~~For RxKotlinFX, perhaps an `Observable#addTo(CompositeObservable)` extension function would be helpful as well to keep it fluent.~~

**Comment by thomasnield** [comment]
I scratched out items that have been taken care of. Now debating another item: 
* Allow "policy options" for each added `Observable`, starting with `Policy.REMOVE_ON_TERMINATE` that will remove an `Observable` from the `CompositeObservable` once it terminates.

**Comment by thomasnield** [comment]
My thoughts have been going all over the place on this. Debating if the returned `Observable` should allow an option to be multicastedt. I need to come up with more robust specifications for this utility, without alienating it to one purpose like being an event bus. It should theoretically work with cold Observables too.

**Comment by thomasnield** [comment]
I think I got it. A `compose()` lambda should be passed as an optional argument to do any manipulations to the aggregated `Observable`. 

``` java
CompositeObservable<Int> selectedId = 
    new CompositeObservable<>(obs -> obs.cache(1).publish().refCount());

```

This will take care of caching, publishing, and any other options you would like to do to the aggregated `Observable` returned from `toObservable()`.

**Comment by thomasnield** [comment]
Released in RxJavaFX 0.2.0. I think this is a wrap.
